presto-main/src/main/java/com/facebook/presto/execution/StageExecutionInfo.java

```diff
@@ -169,6 +171,9 @@ public static StageExecutionInfo create(
                     operatorToStats.compute(id, (k, v) -> v == null ? operatorStats : v.add(operatorStats));
                 }
             }
+            mergedRuntimeStats.mergeWith(taskStats.getRuntimeStats());
+            mergedRuntimeStats.addMetricValue("splitCountPerTask", taskStats.getTotalDrivers());
```

Comment:
Using total drivers to represent split count per task might not be correct I think. It is true for leaf stage tasks that reads from storage (`TableScanOperator`, where each driver consumes one split), but might not be the case when it is reading from another worker (`RemoteSplit`). To properly record this, we might need drivers to report the splits in `DriverStats` and aggregate at task level.

Or we could simply name this `totalDriverPerTask` to keep it consistent.

Reply:
Good point. Renaming it to totalDriverPerTask sounds good. Another option is to only expose this metric for leaf plans which we care most for this metric. Do you know if there is a way to tell if a plan is leaf or not here?